timezone/src/include/timezone/dialogs.rb

```diff
@@ -877,6 +877,20 @@ def TimezoneDialog(args)
           end
         end
 
+        if ret == :hwclock
+          # Keep internal state in sync (Timezone.hwclock <-> @hwclock_s) in
+          # case user enters SetTimeDialog() and sets the time (bsc#1087228).
+
+          @hwclock_s = UI.QueryWidget(Id(:hwclock), :Value) ? :hwclock_utc : :hwclock_localtime
+          SetTimezone(@hwclock_s, timezone, false, false)
+
+          Builtins.y2milestone("hwclock changed to: %1 (%2), diff: %3", @hwclock_s, Timezone.hwclock, Timezone.diff)
```

Comment:
NP: maybe I'd switch to the new logging with `log.info` (do not forget to `include Yast::Logger`)